### PR TITLE
Material Design widgets for pre-Lollipop

### PR DESCRIPTION
**Pull request by Tunous**
Partial implementation of Material Design for pre-Lollipop devices as explained in #391. This doesn't touch settings screen.

Additional change is disabling of drawer toggle animation when drawer is slided since it gets hidden by drawer.

### CODE REVIEW

File: Simplenote/src/main/java/com/automattic/simplenote/BottomSheetDialogBase.java [hunk 1]
```diff
@@ -5,7 +5,6 @@
 import android.support.design.widget.BottomSheetDialog;
 import android.view.WindowManager;
 
-import com.automattic.simplenote.R;
```
Comment (theck13) [inline Simplenote/src/main/java/com/automattic/simplenote/BottomSheetDialogBase.java]: Removing unused imports is always a good practice.  Since this file has no other changes as part of this pull request, it's arguably better to add that to a separate pull request or commit.

File: Simplenote/src/main/java/com/automattic/simplenote/TagsListFragment.java [hunk 2]
```diff
@@ -34,7 +34,7 @@
 
 import java.util.List;
 
-public class TagsListFragment extends ListFragment implements AdapterView.OnItemClickListener, ActionMode.Callback, AbsListView.MultiChoiceModeListener, AdapterView.OnItemLongClickListener, Bucket.Listener<Tag>{
+public class TagsListFragment extends ListFragment implements AdapterView.OnItemClickListener, ActionMode.Callback, AbsListView.MultiChoiceModeListener, AdapterView.OnItemLongClickListener, Bucket.Listener<Tag> {
```
Comment (theck13) [inline Simplenote/src/main/java/com/automattic/simplenote/TagsListFragment.java]: I notice those extra spaces or lack thereof everywhere too.  This might be another thing to put into another pull request or commit in order to keep the functional and style changes distinct.

File: Simplenote/src/main/res/layout/bottom_sheet_info.xml [hunk 3]
```diff
@@ -3,7 +3,9 @@
     xmlns:tools="http://schemas.android.com/tools"
     xmlns:app="http://schemas.android.com/apk/res-auto"
     android:layout_width="match_parent"
-    android:layout_height="match_parent">
+    android:layout_height="match_parent"
+    app:theme="@style/Theme.Simplestyle"
+    >
```
Comment (theck13) [inline Simplenote/src/main/res/layout/bottom_sheet_info.xml]: For style consistency, this can go on the previous line.

**Comment by Tunous** [comment]
Rebased my commits to exclude syntax changes.

**Comment by theck13** [comment]
:shipit:
